### PR TITLE
OH-6 | Reduction plan listing

### PR DESCRIPTION
**Pull request by alvineteubo**

### CODE REVIEW

File: sql/u2g_update/step_x_create_table_for_reductionplan.sql [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  RP_OTHERRATE FLOAT NOT NULL,
+  RP_CREATED_BY VARCHAR(50),
+  RP_CREATED_DATE DATETIME,
+  RP_LAST_MODIFIED_BY VARCHAR(50),
+  RP_LAST_MODIFIED_DATE DATETIME,
+  RP_ACTIVE TINYINT(1) NOT NULL DEFAULT 1,
+  RP_LOCK INT(11) NOT NULL DEFAULT 0,
+  PRIMARY KEY (RP_ID),
+  KEY RP_CREATED_BY (RP_CREATED_BY),
+  KEY RP_LAST_MODIFIED_BY (RP_LAST_MODIFIED_BY)
```
Comment (ArnaudFofou) [inline sql/u2g_update/step_x_create_table_for_reductionplan.sql]: use the same way as Bill to define contraint for foreign key

File: sql/u2g_update/step_x_create_table_for_reductionplan.sql [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+--
+-- Definition of table oh_otherreduction
+--
+
+CREATE TABLE oh_otherreduction (
+  OTR_RP_ID INT(11) NOT NULL,
+  OTR_OTH_ID INT(11) NOT NULL,
+  OTR_REDUCTIONRATE FLOAT NOT NULL,
+  PRIMARY KEY (OTR_RP_ID, OTR_OTH_ID)
+)ENGINE=MyISAM;;
```
Comment (ArnaudFofou) [inline sql/u2g_update/step_x_create_table_for_reductionplan.sql]:
```suggestion
);
```

File: sql/u2g_update/step_x_create_table_for_reductionplan.sql [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+--
+-- Definition of table oh_exampsreduction
+--
+
+CREATE TABLE oh_examsreduction (
+  ER_RP_ID INT(11) NOT NULL,
+  ER_EXA_ID_A VARCHAR(10) NOT NULL,
+  ER_REDUCTIONRATE FLOAT NOT NULL,
+  PRIMARY KEY (ER_RP_ID, ER_EXA_ID_A)
+)ENGINE=MyISAM;;
```
Comment (ArnaudFofou) [inline sql/u2g_update/step_x_create_table_for_reductionplan.sql]:
```suggestion
);
```

File: sql/u2g_update/step_x_create_table_for_reductionplan.sql [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+--
+-- Definition of table oh_medicalsreduction
+--
+CREATE TABLE oh_medicalsreduction (
+  MR_RP_ID INT(11) NOT NULL,
+  MR_MED_ID INT(11) NOT NULL,
+  MR_REDUCTIONRATE FLOAT NOT NULL,
+  PRIMARY KEY (MR_RP_ID, MR_MED_ID)
+)ENGINE=MyISAM;;
```
Comment (ArnaudFofou) [inline sql/u2g_update/step_x_create_table_for_reductionplan.sql]:
```suggestion
);
```

File: sql/u2g_update/step_x_create_table_for_reductionplan.sql [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+--
+-- Definition of table oh_operationsreduction
+--
+
+CREATE TABLE oh_operationsreduction (
+  OPR_RP_ID INT(11) NOT NULL,
+  OPR_OPE_ID_A VARCHAR(10) NOT NULL,
+  OPR_REDUCTIONRATE FLOAT NOT NULL,
+  PRIMARY KEY (OPR_RP_ID, OPR_OPE_ID_A)
+)ENGINE=MyISAM;;
```
Comment (ArnaudFofou) [inline sql/u2g_update/step_x_create_table_for_reductionplan.sql]:
```suggestion
);
```

File: src/main/resources/database.properties.sample [hunk 6]
```diff
@@ -1,3 +1,3 @@
-jdbc.url=jdbc:mariadb://localhost:3306/oh
```
Comment (ArnaudFofou) [inline src/main/resources/database.properties.sample]: remove this file

File: src/test/java/org/isf/reductionplan/TestsReductionplan.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+			assertThat(reductionplan.getDescription()).isEqualTo(description);
+			assertThat(reductionplan.getOperationRate()).isEqualTo(operationRate);
+			assertThat(reductionplan.getMedicalRate()).isEqualTo(medicalRate);
+			assertThat(reductionplan.getExamRate()).isEqualTo(examRate);
+			assertThat(reductionplan.getOtherRate()).isEqualTo(otherRate);
+
+		}
+
+		// Test avec le constructeur
+		@Test
```
Comment (ArnaudFofou) [inline src/test/java/org/isf/reductionplan/TestsReductionplan.java]: please removet that Test here and add it on class Test

File: src/main/java/org/isf/accounting/manager/BillBrowserManager.java
Comment (JantBogard) [inline src/main/java/org/isf/accounting/manager/BillBrowserManager.java:1]: Reset all change

**Review by JantBogard — CHANGES_REQUESTED** [review]

File: src/main/java/org/isf/reductionplan/model/ReductionPlan.java [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+	public int getId() {
+		return id;
+	}
+	public void setId(int id) {
+		this.id = id;
+	}
+
+	public String getDescription() {
+		return description;
+	}
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/reductionplan/model/ReductionPlan.java:112]:
```suggestion
	}
	
```

File: src/main/java/org/isf/reductionplan/model/ReductionPlan.java [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+	public void setDescription(String description) {
+		this.description = description;
+	}
+
+	public double getOperationRate() {
+		return operationRate;
+	}
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/reductionplan/model/ReductionPlan.java:120]:
```suggestion
	}
	
```

File: src/main/java/org/isf/reductionplan/model/ReductionPlan.java [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+	public void setOperationRate(double operationRate) {
+		this.operationRate = operationRate;
+	}
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/reductionplan/model/ReductionPlan.java:124]:
```suggestion
	}
	
```

File: src/main/java/org/isf/reductionplan/model/ReductionPlan.java [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+	public double getMedicalRate() {
+		return medicalRate;
+	}
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/reductionplan/model/ReductionPlan.java:128]:
```suggestion
	}
	
```

File: src/main/java/org/isf/reductionplan/model/ReductionPlan.java [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+	public void setMedicalRate(double medicalRate) {
+		this.medicalRate = medicalRate;
+	}
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/reductionplan/model/ReductionPlan.java:132]:
```suggestion
	}
	
```

File: src/main/java/org/isf/reductionplan/model/ReductionPlan.java [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+	public double getExamRate() {
+		return examRate;
+	}
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/reductionplan/model/ReductionPlan.java:136]:
```suggestion
	}
	
```

File: src/main/java/org/isf/reductionplan/model/ReductionPlan.java [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+	public void setExamRate(double examRate) {
+		this.examRate = examRate;
+	}
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/reductionplan/model/ReductionPlan.java:140]:
```suggestion
	}
	
```

File: src/main/java/org/isf/reductionplan/model/ReductionPlan.java [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+	public double getOtherRate() {
+		return otherRate;
+	}
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/reductionplan/model/ReductionPlan.java:144]:
```suggestion
	}
	
```

File: src/main/java/org/isf/reductionplan/model/ReductionPlan.java [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+		this.otherRate = otherRate;
+	}
+
+	public ReductionPlan(String description, double operationRate, double medicalRate, double examRate, double otherRate) {
+		this.description = description;
+		this.operationRate = operationRate;
+		this.medicalRate = medicalRate;
+		this.examRate = examRate;
+		this.otherRate = otherRate;
+	}
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/reductionplan/model/ReductionPlan.java:156]:
```suggestion
	}
	
```

File: src/main/java/org/isf/reductionplan/model/ReductionPlan.java [hunk 17]
```diff
@@ -0,0 +1,170 @@
...
+
+			c = m * c + id;
+
+			this.hashcode = c;
+		}
+
+		return this.hashcode;
+	}
+}
+
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/reductionplan/model/ReductionPlan.java:193]: remove all 3 space after line 166

File: src/main/java/org/isf/reductionplan/manager/ReductionplanBrowserManager.java [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+	 * Returns the list of {@link ReductionPlan}s in the DB.
+	 * @return the list of {@link ReductionPlan}s
+	 */
+	public List<ReductionPlan> getReductionplan() throws OHServiceException {
+		return ioOperations.getReductionplan();
+	}
+
+	public List<ReductionPlan> getReductionplan(String description) throws OHServiceException {
+		return ioOperations.getReductionPlan(description); // Assurez-vous que description est passé ici
+	}
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/reductionplan/manager/ReductionplanBrowserManager.java:73]:
```suggestion
	}
	
```

File: src/test/java/org/isf/reductionplan/TestsReductionplan.java [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+		// Méthode de vérification des valeurs
+		public void check(ReductionPlan reductionplan) {
+			assertThat(reductionplan.getDescription()).isEqualTo(description);
+			assertThat(reductionplan.getOperationRate()).isEqualTo(operationRate);
+			assertThat(reductionplan.getMedicalRate()).isEqualTo(medicalRate);
+			assertThat(reductionplan.getExamRate()).isEqualTo(examRate);
+			assertThat(reductionplan.getOtherRate()).isEqualTo(otherRate);
+
+		}
+
```
Comment (ArnaudFofou) [inline src/test/java/org/isf/reductionplan/TestsReductionplan.java:44]: remode 2 line of space here

File: src/test/java/org/isf/reductionplan/TestsReductionplan.java [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+		// Méthode de vérification des valeurs
+		public void check(ReductionPlan reductionplan) {
+			assertThat(reductionplan.getDescription()).isEqualTo(description);
+			assertThat(reductionplan.getOperationRate()).isEqualTo(operationRate);
+			assertThat(reductionplan.getMedicalRate()).isEqualTo(medicalRate);
+			assertThat(reductionplan.getExamRate()).isEqualTo(examRate);
+			assertThat(reductionplan.getOtherRate()).isEqualTo(otherRate);
+
```
Comment (ArnaudFofou) [inline src/test/java/org/isf/reductionplan/TestsReductionplan.java:42]:
```suggestion
```

File: .gitignore [hunk 21]
```diff
@@ -9,3 +9,4 @@ logs
 .vscode/
 *.iml
 /.env
+
```
Comment (ArnaudFofou) [inline .gitignore]:
```suggestion
```

File: src/main/java/org/isf/reductionplan/manager/ReductionplanBrowserManager.java [hunk 22]
```diff
@@ -0,0 +1,96 @@
+package org.isf.reductionplan.manager;
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/reductionplan/manager/ReductionplanBrowserManager.java:23]: please add doc title on all the new file

File: src/test/java/org/isf/reductionplan/TestsReductionplan.java [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.isf.reductionplan;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import org.isf.reductionplan.model.ReductionPlan;
+
+class TestsReductionplan{
```
Comment (ArnaudFofou) [inline src/test/java/org/isf/reductionplan/TestsReductionplan.java]:
```suggestion
class TestsReductionplan {
```

**Comment by ArnaudFofou** [comment]
will create new PR because This PR have many work that not correspond the feature

**Pull request closed without merging**